cmd/get-kustomize-build-base/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/urfave/cli/v2"
+)
+
+// see https://git-scm.com/docs/git-diff#_raw_output_format
+const (
+	changeStatusModification = "M"
+	changeStatusChangingType = "T"
+	changeStatusAddition     = "A"
+	changeStatusDeletion     = "D"
+	changeStatusCopy         = "C"
+	changeStatusRenaming     = "R"
+	// not relevant: U and X
+
+	gitDiffStatusColumn = 4
+)
+
+func main() { //go-cov:skip
+	if err := runApp(context.Background(), os.Stdout, os.Stdin, os.Args); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func runApp(ctx context.Context, out io.Writer, in io.Reader, args []string) error {
```

Comment:
Why do we need this function? 
We are splitting the very basic main() function into three components.
If we are not solving any problem this way, wouldn't it be better to just use the convention from the repository?

Reply:
> Why do we need this function? We are splitting the very basic main() function into three components. If we are not solving any problem this way, wouldn't it be better to just use the convention from the repository?

I like my main to do two things:

* Call some other function
* Exit with a corresponding code from that function

I don't like having logic in `main` because It's hard to test: if I want to test anything with args I have to patch `os.Args`, and since I `os.Exit` I can't really test anything about the exit code. Additionally, since I want to `os.Exit` from main I can't expect any `defer`s to be run, so I don't want my main doing any additional setup etc. that needs any resource management since I might run into issues with this